src/main/java/uk/gov/justice/digital/domain/DomainExecutor.java

```diff
@@ -0,0 +1,271 @@
+            final Dataset<Row> transformedDataFrame = applyTransform(refs, table.getTransform());
+
+            logger.info("Apply Violations for " + table.getName() + "...");
+            // Process Violations - we now have a subset
+            final Dataset<Row> postViolationsDataFrame = applyViolations(transformedDataFrame, table.getViolations());
+
+            logger.info("Apply Mappings for " + table.getName() + "...");
+            // Mappings
+            return applyMappings(postViolationsDataFrame, table.getMapping());
+
+        } catch(Exception e) {
+            logger.info("Apply Domain for " + table.getName() + " failed.");
+            handleError(e);
+            throw new DomainExecutorException("Apply Domain for " + table.getName() + " failed.");
+        }
+        finally {
+            logger.info("Apply Domain process for " + table.getName() + " completed.");
+        }
+    }
+
+    public void doFull(final String domainName, final String domainTableName, final String domainOperation) {
+        try {
+            if (domainOperation.equalsIgnoreCase("insert") ||
+                    domainOperation.equalsIgnoreCase("update") ||
+                    domainOperation.equalsIgnoreCase("sync")) {
+                final List<TableDefinition> tables = domainDefinition.getTables();
+                TableDefinition table = tables.stream()
+                        .filter(t -> domainTableName.equals(t.getName()))
+                        .findAny()
+                        .orElse(null);
+                if (table == null) {
+                    logger.error("Table " + domainTableName + " not found");
+                    throw new DomainExecutorException("Table " + domainTableName + " not found");
+                } else {
+                    // TODO no source table and df they are required only for unit testing
+                    final Dataset<Row> df_target = apply(table, null);
+                    saveFull(TableInfo.create(targetRootPath,  domainDefinition.getName(), table.getName()),
+                            df_target, domainOperation);
+                }
+            } else if (domainOperation.equalsIgnoreCase("delete")) {
+                logger.info("domain operation is delete");
+                deleteFull(TableInfo.create(targetRootPath, domainName, domainTableName));
+            } else {
+                logger.error("Unsupported domain operation");
+                throw new UnsupportedOperationException("Unsupported domain operation.");
+            }
+        } catch(Exception e) {
+            handleError(e);
+        } catch (DomainExecutorException e) {
+            logger.error("Domain executor failed: ", e);
+        }
+    }
+
+
+    protected boolean schemaContains(final Dataset<Row> dataFrame, final String field) {
+        return Arrays.asList(dataFrame.schema().fieldNames()).contains(field);
+    }
+
+    protected void handleError(final Exception e) {
```

Comment:
Remove this method since it doesn't do anything useful. Replace with direct usage of `logger.error`. We should not be sending any output to STDOUT as per comments on my earlier review.